src/BaseElement.js

```diff
@@ -302,40 +305,40 @@ class BaseElement extends HTMLElement {
 		}
 	}
 
-	/**
-	 * Context Properties to issue Context Requests and to pull contextual properties into the elements scope
-	 * @return {{}}
-	 */
-	context() {
-		return {};
-	}
-
 	/**
 	 * Defines context on the element based on keys from this.context()
 	 */
 	defineContext() {
-		Object.entries(this.context()).forEach(([key, value]) => {
-			this.dispatch('request-context', { [key]: value }, true);
-		});
 		// define context provider
-		if (this._options.provideContext) {
+		if (Object.values(this._options.propertyOptions).find((option) => option.provideContext === true)) {
+			// at least one property that provides Context
 			this.addEventListener('request-context', this.onRequestContext);
 		}
 	}
 
+	doRequestContext(property, callback) {
```

Comment:
@quarkus do request?! 🤣 since when do we do 😆 This sounds kind of funny since we have never called a method with do in front.